src/utils/appDataClient.ts

```diff
@@ -1,44 +1,129 @@
-import { Hotspot, PocReceiptsV1 } from '@helium/http'
+import Client, {
+  AnyTransaction,
+  Hotspot,
+  Network,
+  PocReceiptsV1,
+  ResourceList,
+} from '@helium/http'
 import { heliumHttpClient } from '@helium/react-native-sdk'
+import Config from 'react-native-config'
 import { fromNow } from './timeUtils'
 import { getAddress } from './secureAccount'
 
-export const submitTxn = async (txn: string) => {
-  return heliumHttpClient.transactions.submit(txn)
-}
+const privateHeliumClient = new Client(
+  new Network({
+    baseURL: Config.HELIUM_SHADOW_URL,
+    version: 1,
+  }),
+)
 
 export const getHotspotDetails = async (address: string): Promise<Hotspot> => {
-  return heliumHttpClient.hotspots.get(address)
+  try {
+    return await privateHeliumClient.hotspots.get(address)
+  } catch (error) {
+    try {
+      return await heliumHttpClient.hotspots.get(address)
+    } catch (err) {
+      throw new Error(
+        `Helium API call is failed while getting the hotspot details: ${err}`,
+      )
+    }
+  }
+}
+
+export const submitTxn = async (txn: string) => {
+  try {
+    return await heliumHttpClient.transactions.submit(txn)
+  } catch (err) {
+    throw new Error(
+      `Helium API call is failed while submitting the transaction: ${err}`,
+    )
+  }
 }
 
 export const getHotspots = async () => {
   const address = await getAddress()
   if (!address) return []
 
-  const newHotspotList = await heliumHttpClient.accounts
-    .fromAddress(address)
-    .hotspots.list()
+  try {
+    const newHotspotList = await privateHeliumClient.accounts
+      .fromAddress(address)
+      .hotspots.list()
+
+    return newHotspotList.takeJSON(10000)
+  } catch (error) {
+    try {
+      const newHotspotList = await heliumHttpClient.accounts
+        .fromAddress(address)
+        .hotspots.list()
 
-  return newHotspotList.takeJSON(10000)
+      return newHotspotList.takeJSON(10000)
+    } catch (err) {
+      throw new Error(
+        `Helium API call is failed while getting the list of hotspots: ${err}`,
+      )
+    }
+  }
 }
 
 export const getAccount = async (address?: string) => {
   if (!address) return
 
-  const { data } = await heliumHttpClient.accounts.get(address)
-  return data
+  try {
+    const { data } = await privateHeliumClient.accounts.get(address)
+    return data
+  } catch (error) {
+    try {
+      const { data } = await heliumHttpClient.accounts.get(address)
+      return data
+    } catch (err) {
+      throw new Error(
+        `Helium API call is failed while getting the account information: ${err}`,
```

Comment:
Helium API request failed while trying to retrieve account information for {address}.